eclair-core/src/main/scala/fr/acinq/eclair/io/Switchboard.scala

```diff
@@ -69,10 +69,15 @@ class Switchboard(nodeParams: NodeParams, router: ActorRef, watcher: ActorRef, r
         case None => sender ! Status.Failure(new RuntimeException("no connection to peer"))
       }
 
-    case authenticated: PeerConnection.Authenticated =>
-      // if this is an incoming connection, we might not yet have created the peer
-      val peer = createOrGetPeer(authenticated.remoteNodeId, offlineChannels = Set.empty)
-      authenticated.peerConnection ! PeerConnection.InitializeConnection(peer)
+    case p: PeerConnection.ConnectionReady =>
+      // we create a peer if it doesn't exist
+      val peer = createOrGetPeer(p.remoteNodeId, offlineChannels = Set.empty)
+      peer forward p
+
+    case p: Peer.PeerMessage =>
+      // we create a peer if it doesn't exist
```

Comment:
We should never have to create one here, right?
Are there unwanted side-effects if we do (some assumption each of the FSMs make about the other FSMs)?
Shouldn't we reject those messages if we don't have a peer actor to handle it?

It feels to me that the peer FSM already have to know about the other FSMs and assume some of their behavior, and this change is making it slightly worse...

Reply:
> We should never have to create one here, right?

Correct, this is only needed in the handler for `PeerConnection.ConnectionReady`.

> It feels to me that the peer FSMs already have to know about the other FSMs and assume some of their behavior, and this change is making it slightly worse...

There has to be an interface contract, even if it less explicit with (untyped) actors. If anything the interface is a bit simpler now. But let's revert this and we can revisit later.